src/main/java/org/myrobotlab/service/Cron.java

```diff
@@ -1,44 +1,73 @@
 package org.myrobotlab.service;
 
 import java.io.Serializable;
-import java.util.ArrayList;
+import java.util.Map;
+import java.util.UUID;
 
-import org.myrobotlab.codec.CodecUtils;
 import org.myrobotlab.framework.Service;
 import org.myrobotlab.logging.Level;
 import org.myrobotlab.logging.LoggerFactory;
 import org.myrobotlab.logging.Logging;
 import org.myrobotlab.logging.LoggingFactory;
+import org.myrobotlab.service.config.CronConfig;
 import org.slf4j.Logger;
 
 import it.sauronsoftware.cron4j.Scheduler;
 
 /**
- * Cron - This is a cron based service that can execute a "task" at some point
- * in the future such as "invoke this method on that service"
+ * Cron - This is a cron based service that can execute a "task".
  * 
- * FIXME - the common cron notation is kind of nice - but this thing doesn't do
- * more than Service.addTask
- * 
- * FIXME - make a purge &amp; delete DUH !
- *
  */
 public class Cron extends Service {
 
   public static class Task implements Serializable, Runnable {
+
     private static final long serialVersionUID = 1L;
-    transient Cron myService;
+    /**
+     * cron pattern for this task
+     */
     public String cronPattern;
-    public String name;
-    public String method;
+
+    /**
+     * data parameters to invoke
+     */
     public Object[] data;
 
-    public Task(Cron myService, String cronPattern, String name, String method) {
-      this(myService, cronPattern, name, method, (Object[]) null);
+    /**
+     * unique hash the scheduler uses (only)
+     */
+    transient public String hash;

```

Comment:
its overly smart .. it should be defined here, it should probably just be part of CronConfig...  "if" that was the case, it would be public.  So next refactor I'll make it part of config directly.